Source/CombatExtended/CombatExtended/Verbs/Verb_LaunchProjectileCE.cs

```diff
@@ -348,9 +348,157 @@ public virtual void ShiftTarget(ShiftVecReport report, bool calculateMechanicalO
                     }
                     else if (currentTarget.Thing is Pawn)
                     {
-                        // Aim for center of mass on an exposed target
-                        targetRange.min = victimVert.BottomHeight;
+                        targetRange.min = victimVert.BottomHeight;
                         targetRange.max = victimVert.MiddleHeight;
+
+                        bool AppropiateAimMode = CompFireModes?.CurrentAimMode != AimMode.SuppressFire;
+
+                        bool flagautomatic = (CompFireModes?.target_mode != TargettingMode.automatic);
+
+                        if ( (CasterPawn?.Faction ?? Caster.Faction) == Faction.OfPlayer && flagautomatic)
+                        {
+                            bool flag1 = ((ShootingAccuracy >= 2.2f) | (Caster is Building_TurretGunCE));
+
+                           
+
+                            if (flag1)
+                            {
+                                if (AppropiateAimMode)
+                                {
+                                    if (CompFireModes?.target_mode == TargettingMode.head)
+                                    {
+                                        // Aim upper thoraxic to head
+                                        targetRange.min = victimVert.MiddleHeight;
+                                        targetRange.max = victimVert.Max;
+                                    }
+                                    else if (CompFireModes?.target_mode == TargettingMode.legs)
+                                    {
+                                        // Aim legs to lower abdomen
+                                        targetRange.min = victimVert.Min;
+                                        targetRange.max = victimVert.BottomHeight;
+                                    }
+                                    else
+                                    {
+                                        // Aim center mass
+                                        targetRange.min = victimVert.BottomHeight;
+                                        targetRange.max = victimVert.MiddleHeight;
+                                    }
+                                }
+                            }
+                        }
+                        else
+                        {
+                            var Victim = (Pawn)CurrentTarget.Thing;
+
+                            targetRange.min = victimVert.BottomHeight;
+                            targetRange.max = victimVert.MiddleHeight;
+

```

Comment:
Might be best to push this to its own function.  Tag it inline so the performance impact is reduced.